User type: Super admin

DXCanas opened the pull request:
<!--
 1. Following guidance below, replace …'s with your own words
 2. After saving the PR, tick of completed checklist items
 3. Skip checklist items that are not applicable or not necessary
 4. Delete instruction/comment blocks
-->

### Summary
<!--
 * description of the change
 * manual verification steps performed
 * screenshots if the PR affects the UI
-->
Addressing user admin updates in [this prototype](https://www.figma.com/file/KxjIhoBljRKy71SARf1h7dO0/0.11-superuser-clarification?node-id=182%3A0)
facility user list as regular facility admin
![facility user list as regular facility admin](https://user-images.githubusercontent.com/9877852/45792862-70bc8080-bc44-11e8-963e-67f111687230.png)
facility user list as super admin
![facility user list as super admin](https://user-images.githubusercontent.com/9877852/45792884-892c9b00-bc44-11e8-9f18-ddc90e9a824c.png)
super admin editing self in user edit
![super admin editing self in user edit](https://user-images.githubusercontent.com/9877852/45792902-a19cb580-bc44-11e8-8fd5-76daf0ee2657.png)
super admin editing other super admin in user edit
![super admin editing other super admin in user edit](https://user-images.githubusercontent.com/9877852/45792915-abbeb400-bc44-11e8-973a-4c1989eb5555.png)
super admin editing self in device permissions
![super admin editing self in device permissions](https://user-images.githubusercontent.com/9877852/45792984-e88aab00-bc44-11e8-9431-bbfefe2752e4.png)
super admin editing other super admin in device permissions
![super admin editing other super admin in device permissions](https://user-images.githubusercontent.com/9877852/45792953-cee96380-bc44-11e8-88e2-9d969879bb51.png)
![super admin device permissions view with tooltip](https://user-images.githubusercontent.com/9877852/45792999-f809f400-bc44-11e8-9bf5-29183c6d0dab.png)
super admin profile view
![super admin profile view](https://user-images.githubusercontent.com/9877852/45793017-08ba6a00-bc45-11e8-83ae-af4ec7bb5bc4.png)
content-enabled admin profile view
![content-enabled admin profile view](https://user-images.githubusercontent.com/9877852/45793119-72d30f00-bc45-11e8-9cea-c67823d2bfaf.png)
super admin user dropdown view
![super admin user dropdown view](https://user-images.githubusercontent.com/9877852/45793027-11ab3b80-bc45-11e8-93da-1f53892475fc.png)
super admin stripping other super admin of superness
![super admin stripping other super admin of superness](https://user-images.githubusercontent.com/9877852/45793037-1e2f9400-bc45-11e8-812f-533555a32f25.png)
![super admin stripping other super admin of superness WITH TOOLTIP](https://user-images.githubusercontent.com/9877852/45793070-39020880-bc45-11e8-91ae-ee433aaa8085.png)

### Reviewer guidance
<!--
 * how can a reviewer test these changes?
 * are there any risky areas that deserve extra testing
-->

Used regex `(:\s['"].*superuser.*['"])` to find instances of superuser in `*.js` and `*.vue` files. Make sure I didn't miss any user-facing strings. Replaced most of them with new `DisplayUserType` component. Informing that with new `UserType` util.

There is no more "role" or "kind". Only "type.

Modified the `EditUserModal` a good amount. Probably in need of testing.

### References
<!--
 * references to related issues and PRs
 * links to mockups or specs for new features
 * links to the diffs for any dependency updates, e.g. in iceqube or the perseus plugin
-->

- [Design Trello Card](https://trello.com/c/0mgXeeS5/196-clarify-super-admin-in-user-management-tab)
- Resolves #4211
----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If there are any front-end changes, before/after screenshots are included
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

DXCanas committed:
Setting up user info section.

DXCanas committed:
Adding facility name to device permissions page.

DXCanas committed:
Laying out user information in a table rather than dl.

DXCanas committed:
Wrapping strings.

DXCanas committed:
Adding superuser description to page without modifying KCheckbox

DXCanas committed:
Update/stay on page after modifying user device permissions
Adding dummy code to display user kind in user device permissions page
Superuser->Super admin in user permissions page
Superuser->Super admin in permissions icon tooltip
Adding superuser to getUserKind state getter
Tweaking styles of user info in user device permissions

DXCanas committed:
Adding snackbar confirmation to permission change.

DXCanas committed:
Moving UserPermissionsPage to folder structure

DXCanas committed:
Creating non-styled user type component.
Inspired by UserRole, FacilityUsers handler.

DXCanas committed:
Breaking component up to include a util for use in logic.

DXCanas committed:
Integrating UserType and UserTypeDisplay in Facility/UserTable

DXCanas committed:
displayLearner->omitLearner to match pre-existing API.

DXCanas committed:
Using edited user's facility rather than logged in user's facility in User Device permissions.

DXCanas committed:
Replacing role label in navbar.

DXCanas committed:
Updating type label in profile page.

DXCanas committed:
Changing default behavior of UserTypeDisplay to display learners

DXCanas committed:
Minor refactor and simplification of PermissionIcon to use UserTypeDisplay.
Adding position to ui-tooltip for better visibility on slim displays.

DXCanas committed:
Having role-only changes close modal.
Resolves #4211

DXCanas committed:
Adding conditional superuser statement to EditUserModal.

DXCanas committed:
superuser -> super admin in remaining strings.

DXCanas committed:
Converting AuthMessage user type strings to lowercase to match string spec.

indirectlylit commented:
looks like all the proper nouns got made lower-case

indirectlylit commented on kolibri/plugins/facility_management/assets/src/views/UserPage/EditUserModal.vue:
```diff
@@ -32,26 +32,43 @@
-        :value="false"
-        v-model="classCoachIsSelected"
+      <span class="super-admin-description">
+        Go to Device permissions to change this
```
`$tr`

christianmemije replied:
Maybe link to this page too?

DXCanas replied:
> Maybe link to this page too?

On the map, see: https://www.figma.com/file/KxjIhoBljRKy71SARf1h7dO0/0.11-superuser-clarification?node-id=0%3A1

indirectlylit commented:
actually .. looks like we're not consistent with capitalization anyway. @jtamiace @radinamatic  I think we decided to use proper nouns when referring to permissions and user types, right?

E.g.

> Make a new Coach user for the coach.

?

christianmemije commented on kolibri/plugins/device_management/assets/src/views/UserPermissionsPage/index.vue:
```diff
@@ -190,23 +233,29 @@
       cancelButton: 'Cancel',
       devicePermissions: 'Device permissions',
       devicePermissionsDetails: 'Can import and export content channels',
+      documentTitle: "{ name }'s Device Permissions",
```
We should also use this string for the h1 or the app bar title?

DXCanas replied:
Hmm this wasn't specced in [the mocks](https://www.figma.com/file/KxjIhoBljRKy71SARf1h7dO0/0.11-superuser-clarification?node-id=0%3A1). Using this for either of those wouldn't match the spec.

christianmemije commented on kolibri/plugins/device_management/assets/src/views/UserPermissionsPage/index.vue:
```diff
@@ -221,12 +270,36 @@
+
+  .checkbox-description {
+    // visual estimate, supposed to line up with checkbox label
+    margin: 0;
```
Maybe this instead
```css
margin: 0px 0 0 50px;
padding: 0;
```

DXCanas replied:
Is the benefit here that we're explicitly defining padding? (I got out of the habit of one-lined margin defs because stylint used to break using them lol).

christianmemije commented on kolibri/plugins/device_management/assets/src/views/UserPermissionsPage/index.vue:
```diff
@@ -221,12 +270,36 @@
   }
 
   .permissions-icon {
-    padding-right: 8px;
```
A left margin would be nice.

christianmemije commented on kolibri/plugins/facility_management/assets/src/views/UserTable.vue:
```diff
@@ -177,7 +179,15 @@
+    color: $core-bg-light;
+    white-space: nowrap;
+    background-color: $core-text-annotation;
+    border-radius: 0.5em;
```
Let's increase the line-height and make to make the pill thiccer.

DXCanas replied:
Not specced!!

christianmemije commented:
Looking good.

indirectlylit commented on kolibri/plugins/device_management/assets/src/views/UserPermissionsPage/index.vue:
```diff
@@ -1,35 +1,66 @@
+            <th scope="row">
+              {{ $tr('usernameLabel') }}
+            </th>
+            <td> {{ user.username }} </td>
```
Minor comment, but in general throughout the codebase we would write

```html
<td>{{ user.username }}</td>
```

rather than

```html
<td> {{ user.username }} </td>
```

In addition to codebase consistency, once in a while the extra whitespace actually has an unexpected style or layout impact.

The extra inline whitespace is a little harder to spot than the super obvious case above e.g.

```html
<th scope="row">
  {{ $tr('usernameLabel') }}
</th>
```

DXCanas replied:
What would the ideal fix be for extra inline whitespace?

jtamiace commented:
> I think we decided to use proper nouns when referring to permissions and user types, right?

When did this decision happen? I thought we didn't want to do this

DXCanas committed:
Re-styling checkbox description in UserPermissionsPage.
Explicitly defining padding, one-lining margin definition.

DXCanas committed:
Adding margin to permission icon

indirectlylit commented:
> When did this decision happen? I thought we didn't want to do this

I must be confused. let's make sure we all get on the same page later

DXCanas committed:
Removing unintended whitespace in string injection

DXCanas committed:
Using session getters rather than custom checks in EditUserModal

DXCanas committed:
Adding link to device permissions page for user.
Making text conditional based on whether or not user is superuser.

DXCanas committed:
Can't change self to learner anymore, removing that case.

DXCanas committed:
Remove need for UserKinds in template.

DXCanas committed:
Fixing superuser edits within EditUserModal.
Stop requiring `role` on updateUser.
Cleaning up role kind logic newKind -> newType.

DXCanas committed:
Adding "device permissions" description to profile page for super admins.

DXCanas commented:
@indirectlylit @jtamiace 
Can we resolve this?
![auth message](https://user-images.githubusercontent.com/9877852/45793265-29cf8a80-bc46-11e8-9ccb-7597c8d6a55c.png)

indirectlylit commented:
@DXCanas please go with what ever @jtamiace  spec'd :)

if we have occasional inconsistencies in capitalization it's not the end of the world - this string audit is still a huge improvement

DXCanas committed:
Merge remote-tracking branch 'LE/develop' into superuser-ux-updates

indirectlylit committed:
fix test

indirectlylit commented on kolibri/plugins/device_management/assets/src/views/UserPermissionsPage/index.vue:
```diff
@@ -1,35 +1,56 @@
+        <PermissionsIcon permissionType="SUPERUSER" class="permissions-icon" />
 
-      <hr>
+        <ul :class="['checkbox-description', {disabled: superuserDisabled}]">
```
for unconditionally applied classes, simply add them to a `class` attribute to be clear that they're static:

```html
<ul class="checkbox-description" :class="{disabled: superuserDisabled}">
```

indirectlylit committed:
update device permissions table styling

indirectlylit committed:
clean up class definitions

indirectlylit commented on kolibri/plugins/facility_management/assets/src/views/UserPage/EditUserModal.vue:
```diff
@@ -267,4 +308,19 @@
+
+  .super-admin-description {
+    font-size: 12px;
+    color: $core-text-annotation;
```
non-standard link styling

indirectlylit committed:
remove non-standard link styling

indirectlylit reviewed (approved):
looks good! left a couple comments and pushed changes to address them. Will merge when tests pass
